src/clj/puppetlabs/services/jruby/jruby_core.clj

```diff
@@ -183,6 +184,12 @@
   [jruby-instance :- (schema/pred jruby-schemas/jruby-instance?)]
   @(:state jruby-instance))
 
+(schema/defn register-event-handler
+  "Register the callback function by adding it to the event callbacks atom."
+  [event-callbacks :- Atom
+   callback-fn :- IFn]
+  (swap! event-callbacks conj callback-fn))
```

Comment:
In this case I do think it makes sense to pass it in, since it's used for updating the atom.

However, I am actually not entirely sure of the use case for this, and whether it's still needed. I have only a vague idea of what the use case is (having multiple services be able to register event handlers), but probably a more concrete use case would help me understand.

For now, it seems like users would pass the atom into `create-pool` in the `init` of whatever service they are using the pool from, and then if the wanted to register event handlers from multiple other services they would create a `register-event-handler` service function in that service, and then have other services use that? Or would it make more sense for the `pool-manager-service` to still have a `register-event-handler` service function and handle creation/adding to the pool-context of the `event-callbacks` atom itself?